app/assets/javascripts/discourse/app/components/topic-map/topic-map-summary.gjs

```diff
@@ -1,162 +1,421 @@
+import didInsert from "@ember/render-modifiers/modifiers/did-insert";
+import { service } from "@ember/service";
 import { htmlSafe } from "@ember/template";
 import { gt } from "truth-helpers";
+import ConditionalLoadingSpinner from "discourse/components/conditional-loading-spinner";
 import DButton from "discourse/components/d-button";
-import RelativeDate from "discourse/components/relative-date";
+import TopicMapLink from "discourse/components/topic-map/topic-map-link";
 import TopicParticipants from "discourse/components/topic-map/topic-participants";
+import TopicViews from "discourse/components/topic-map/topic-views";
+import TopicViewsChart from "discourse/components/topic-map/topic-views-chart";
+import avatar from "discourse/helpers/bound-avatar-template";
 import number from "discourse/helpers/number";
 import slice from "discourse/helpers/slice";
+import { ajax } from "discourse/lib/ajax";
+import { emojiUnescape } from "discourse/lib/text";
+import dIcon from "discourse-common/helpers/d-icon";
 import i18n from "discourse-common/helpers/i18n";
-import { avatarImg } from "discourse-common/lib/avatar-utils";
+import I18n from "discourse-i18n";
+import DMenu from "float-kit/components/d-menu";
+import and from "truth-helpers/helpers/and";
+import lt from "truth-helpers/helpers/lt";
+import not from "truth-helpers/helpers/not";
+
+const TRUNCATED_LINKS_LIMIT = 5;
+const MIN_POST_READ_TIME = 4;
+
+export default class SimpleTopicMapSummary extends Component {
+  @service site;
+  @service siteSettings;
+  @service mapCache;
+
+  @tracked allLinksShown = false;
+  @tracked mostLikedPosts = [];
+  @tracked views = [];
+  @tracked loading = true;
 
-export default class TopicMapSummary extends Component {
   get linksCount() {
     return this.args.topicDetails.links?.length ?? 0;
   }
 
-  get createdByUsername() {
-    return this.args.topicDetails.created_by?.username;
+  get topicLinks() {
+    return this.args.topicDetails.links;
   }
 
-  get lastPosterUsername() {
-    return this.args.topicDetails.last_poster?.username;
+  get linksToShow() {
+    return this.allLinksShown
+      ? this.topicLinks
+      : this.topicLinks.slice(0, TRUNCATED_LINKS_LIMIT);
   }
 
-  get toggleMapButton() {
-    return {
-      title: this.args.collapsed
-        ? "topic.expand_details"
-        : "topic.collapse_details",
-      icon: this.args.collapsed ? "chevron-down" : "chevron-up",
-      ariaExpanded: this.args.collapsed ? "false" : "true",
-      ariaControls: "topic-map-expanded__aria-controls",
-      action: this.args.toggleMap,
-    };
+  @action
+  showAllLinks() {
+    this.allLinksShown = true;
   }
 
   get shouldShowParticipants() {
     return (
-      this.args.collapsed &&
-      this.args.topic.posts_count > 2 &&
-      this.args.topicDetails.participants &&
-      this.args.topicDetails.participants.length > 0
+      this.args.topic.posts_count >= 10 &&
+      this.args.topicDetails.participants?.length >= 2 &&
```

Comment:
We might want to extract the two magic numbers (10 and 2) into properly named constants?